ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/drop/AbstractDropPartitionAnalyzer.java

```diff
@@ -90,13 +90,14 @@ protected void analyzeCommand(TableName tableName, Map<String, String> partition
         throw se;
       }
     }
+    validateAlterTableType(table, AlterTableType.DROPPARTITION, expectView());
```

Comment:
Is there any particular reason for moving this validation here?

Reply:
Ah, yes, thanks for asking, I wanted to provide an explanatory comment. So I think the current code flow is not perfect, because `ALTER TABLE tbl DROP PARTITION` commands should be disabled completely for non-native tables (that's what the `validateAlterTableType` checks). 

However, right now the logic which parses and validates the partition clause part of the syntax tree (`ParseUtils.getFullPartitionSpecs`) runs first, therefore if you specify a partition which does not exist, e.g. `ALTER TABLE tbl DROP PARTITION('department'='does-not-exist')` on an Iceberg or HBase or whatever table, then you'll get an exception saying the `partition does not exist` instead of what I think we'd expect, which is `ALTER TABLE DROP PARTITION is not allowed for non-native tables`

Reply:
Thanks for the explanation.